test/support.ts

```diff
@@ -441,6 +443,18 @@ export function expectsql(
     }
   }
 
+  if (typeof query === 'function') {
+    try {
+      query = query();
+    } catch (error: unknown) {
+      if (!(error instanceof Error)) {
+        throw new TypeError('expectsql: function threw something that is not an instance of Error.');
```

Comment:
You can throw anything, even primitives
![image](https://user-images.githubusercontent.com/1280915/191942352-e7968d11-0fc5-439d-b6fd-887540a9fc05.png), hence why the type of caught errors is `unknown` (since TS 4.4 https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#use-unknown-catch-variables)

The rest of `expectsql` expects errors to be instances of Error, and everything we throw in our codebase should inherit from `Error` (because throwing something else is bad practice), hence why we assert that the thrown error is an instance of Error

Reply:
> You can throw anything, even primitives ![image](https://user-images.githubusercontent.com/1280915/191942352-e7968d11-0fc5-439d-b6fd-887540a9fc05.png), hence why the type of caught errors is `unknown` (since TS 4.4 https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#use-unknown-catch-variables)
> 
> The rest of `expectsql` expects errors to be instances of Error, and everything we throw in our codebase should inherit from `Error` (because throwing something else is bad practice), which is why we assert that the thrown error is an instance of Error

Thanks for the explanation, lets merge this~!!